Apps/Src/Print_Queue.c

```diff
@@ -0,0 +1,69 @@
+/* Copyright (c) 2018-2022 UT Longhorn Racing Solar */
+/** Print_Queue.c
+ * Queue that holds all characters to be printed that Task_Print needs to send to UART.
+ */
+
+#include "Print_Queue.h"
+#include "RTOS_BPS.h"
+
+#define FIFO_TYPE char
+#define FIFO_SIZE (128)
+#define FIFO_NAME Print_Fifo
+#include "fifo.h"
+
+static Print_Fifo_t printFifo;
+static OS_SEM printFifo_Sem;
+
+/**
+ * @brief Initializes the print queue
+ * @param none
+ * @return none
+ */
+void Print_Queue_Init() {
+    Print_Fifo_renew(&printFifo);
+    RTOS_BPS_SemCreate(&printFifo_Sem, "printSem", 0);
+}
+
+bool New_Line_Check();
```

Comment:
move entire helper function up or define this at the top of the file and also make static, check style guide for more info:
https://bps.readthedocs.io/en/latest/How_To/coding-style.html#private-functions